json.go

```diff
@@ -6,7 +6,8 @@ import (
 	"net/http"
 )
 
-func respondWithError(w http.ResponseWriter, code int, msg string) {
+func respondWithError(w http.ResponseWriter, code int, msg string, err error) {
+	log.Println(err)
```

Comment:
```suggestion
func respondWithError(w http.ResponseWriter, code int, msg string, logErr error) {
	if logErr != nil {
		log.Println(logErr)
	}
```
I think we want to be able to opt out of logging expected errs (err no rows for example)

Maybe even unify it so that it only logs (regardless of status code) if `logErr` is not nil? Though splitting it up is fine too.

---

Though if the code isn't changed in the course anyway and we're just doing this on non-nil errors... I'm overthinking it

Reply:
The way I see it is that expected errors would be handled in the handler code with a different call to `respondWithError` with a different message/logErr

```go
		if errors.Is(err, ErrNoRows) {
			respondWithError(...)
			return
		}
		if err != nil {
			respondWithError(...)
			return
		}
```

Does this make sense?